Deparser: Add parens around TypeCast in IndexElem

lfittl opened the pull request:
We were incorrectly omitting these parentheses, as well as missing a space outputting "CREATE INDEX ON t (col::typecastops_class)" instead of "CREATE INDEX ON t ((col::typecast) ops_class)".

Fixes #213

lfittl committed:
Deparser: Add parens around TypeCast in IndexElem

We were incorrectly omitting these parentheses, as well as missing a space
outputting "CREATE INDEX ON t (col::typecastops_class)" instead of
"CREATE INDEX ON t ((col::typecast) ops_class)".

Fixes #213

lfittl committed:
Deparser: Treat type casts inside "func_expr_windowless" as functions

This is mainly in use for RangeFunction, but the logic in the grammar
here is more broad, allowing type casts to show up in places where
a function call would be expected. Fix by always using the "cast(..)"
syntax in such cases.

msepga reviewed (approved):
:+1: Will rebase #225 such that eef3f76a is dropped in favor of this.

lfittl committed:
Deparser: Add test for casts using function syntax in FROM

This covers the case where deparseFuncExprWindowless needs to handle
a TypeCast and output like a function call, since the other cast syntax
is not valid in FROM.
